pandera/backends/pyspark/container.py

```diff
@@ -102,6 +106,7 @@ def _data_checks(
 
         return check_obj
 
+    @cache_check_obj()
```

Comment:
The decorator was added to PySpark's DataFrameSchemaBackend, that is the funnel point of validations. It is also responsible for dispatching the validation steps.

Reply:
@filipeo2-mck, I assume this implementation assumes the user either wants to cache or persist all DataFrames in the session? Would it make sense to allow for more granular control, i.e letting the user specify if they want to cache a specific DataFrame or not?

Reply:
Exactly, either way it's a massive improvement and we can consider more granular control in other PRs!